rero_ils/modules/stats_cfg/api.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2023 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Statistics configuration for report."""
+from functools import partial
+
+from rero_ils.modules.api import IlsRecord, IlsRecordsIndexer, IlsRecordsSearch
+from rero_ils.modules.fetchers import id_fetcher
+from rero_ils.modules.minters import id_minter
+from rero_ils.modules.providers import Provider
+from rero_ils.modules.stats.api import StatsSearch
+from rero_ils.modules.stats_cfg.extensions import StatConfigDataExtension
+from rero_ils.modules.stats_cfg.models import StatCfgIdentifier, \
+    StatCfgMetadata
+
+# provider
+StatCfgProvider = type(
+    'StatCfgProvider',
+    (Provider,),
+    dict(identifier=StatCfgIdentifier, pid_type='stacfg')
+)
+# minter
+stat_cfg_id_minter = partial(id_minter, provider=StatCfgProvider)
+# fetcher
+stat_cfg_id_fetcher = partial(id_fetcher, provider=StatCfgProvider)
+
+
+class StatsConfigurationSearch(IlsRecordsSearch):
+    """Statistics configuration search."""
+
+    class Meta:
+        """Search only on stats_cfg index."""
+
+        index = 'stats_cfg'
+        doc_types = None
+        fields = ('*',)
+        facets = {}
+
+        default_filter = None
+
+    def get_statistics_configuration(self, status='enabled'):
```

Comment:
I didn't find the place where this function is used (but my eyes are certainly not perfect).
If not used, remove it.

Reply:
I will remove it from this PR. We need it in the another PR for the statistics, I will add it there.